scheduler/views.py

```diff
@@ -229,3 +229,31 @@ def create(request: HttpRequest) -> HttpResponse:
             return render(request, 'index.html', context=context)
         return render(request, 'edit.html', context={"form": form})
     return render(request, 'edit.html', context={"form": EditForm()})
+
+
+def professors(request: HttpRequest) -> HttpResponse:
+    """Render the professors page"""
+    professors_list = Professor.objects.all()
+    context = {'professors': professors_list, 'form': SelectProfessorForm()}
+    if request.method == 'POST':
+        if 'choose' in request.POST:
+            form = SelectProfessorForm(request.POST)
+            if form.is_valid():
+                professor = form.cleaned_data['professor']
+                email = professor.email
+                if not email:
+                    email = "Noemail"
+                context = {'professors': professors_list, 'form': form, 'email': email}
+        elif 'save' in request.POST:
+            form = SelectProfessorForm(request.POST)
+            if form.is_valid():
+                email = request.POST.get('email')
+                professor = form.cleaned_data['professor']
+                professor.email = email
+                professor.save()
+                context = {'professors': professors_list, 'form': SelectProfessorForm()}
```

Comment:
Wydaje mi się, że tu brakuje elsa, bo jak się wpisze email costam@costam to rzuca ValidationError.

Reply:
Dodałam sprawdzanie w html. Teraz po prostu nie przyjmuje podobnych maili. I enter już wywołuje button save:)